include/posix.c

```diff
@@ -726,7 +729,10 @@ void * fzE_file_open(char * file_name, int64_t * open_results, file_open_mode mo
   // make sure no fork is done while we open file
   fzE_lock();
 
-  FILE * fp = fopen(file_name, mode==FZ_FILE_MODE_READ ? "rb" : "a+b");
+  FILE *fp = fopen(file_name, "r+b");
+  if (!fp) {
+    fp = fopen(file_name, "w+b");
+  }
```

Comment:
so now, writing is no longer possible at all unless they file doesn't exist in the first place?

Reply:
Problem is, if we just use `w+b` then file is truncated. If we use `a+b` then any writes will always be at end of file.
This way we first try `r+b` which opens read/write. Then if file does not exist we try again with `w+b`, this then creates the file.

Reply:
With this change truncation should never happen. See test.